apps/definition_kafka/lib/kafka/topic.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Kafka.Topic do
+  use Definition, schema: Kafka.Topic.V1
+
+  defstruct version: 1,
+            pid: nil,
+            endpoints: nil,
+            name: nil,
+            partitions: 1,
+            partitioner: :default,
+            key_path: []
+
+  def on_new(struct) do
+    endpoints =
+      struct.endpoints
+      |> List.wrap()
+      |> Enum.map(&fix_endpoint/1)
+
+    struct
+    |> Map.put(:endpoints, endpoints)
+    |> Map.update!(:partitioner, &fix_partitioner/1)
+    |> Ok.ok()
```

Comment:
Why not handle fix_endpoints and fix_partitioner in the same way? can't both be "fixed" via the `Map.update!` method and include the list wrapping as part of the fix endpoint function as needed?

Reply:
This is code largely copied over from the Kafka writer modules. I'll shore it up later.